### PR TITLE
Release 1.2.1 to devel

### PR DESCRIPTION
**Pull request by pconesa**
Hope this is the latest PR migrating changes from release-1.2.1

Please there are changes that have to go to other repos due to pluginization:

Please, @dmaluenda ....have a look at them for xmipp and viz ones
There is also a change for relion plugin: @azazellochg ? 
And a motioncor2 test

### CODE REVIEW

File: pyworkflow/em/packages/xmipp3/protocol_particle_pick_pairs.py [hunk 1]
```diff
@@ -81,7 +79,7 @@ def convertInputStep(self):
 
     def launchParticlePickGUIStep(self):
         process = launchTiltPairPickerGUI(self.micsFn, self._getExtraPath(),
-                                          self, memory='%dg' % self.memory.get())
+                                          self)
```
Comment (dmaluenda) [inline pyworkflow/em/packages/xmipp3/protocol_particle_pick_pairs.py]: Take this changes to the [Xmipp plugin](https://github.com/I2PC/scipion-em-xmipp)

File: software/em/xmipp/applications/programs/movie_optical_alignment_cpu/movie_optical_alignment_cpu.cpp [hunk 2]
```diff
@@ -346,7 +346,7 @@ class ProgOpticalAligment: public XmippProgram
         {
             gain.read(fnGain);
             applyWindow(gain());
-            gain() = 1.0/gain();
+            //Do not divide by the gain: gain() = 1.0/gain();
```
Comment (dmaluenda) [inline software/em/xmipp/applications/programs/movie_optical_alignment_cpu/movie_optical_alignment_cpu.cpp]: Take this changes to the [xmipp repo](https://github.com/I2PC/xmipp).

File: software/em/xmipp/java/src/xmipp/viewer/particlepicker/training/AutopickRunnable.java [hunk 3]
```diff
@@ -68,7 +68,7 @@ public void run()
 					picker.loadMicrographData(micrograph);
 			}
 
-			boolean updateAllMicrographs = isGeneric && ((GenericClassifier) classifier).doPickAll();
+			final boolean updateAllMicrographs = isGeneric && ((GenericClassifier) classifier).doPickAll();
```
Comment (dmaluenda) [inline software/em/xmipp/java/src/xmipp/viewer/particlepicker/training/AutopickRunnable.java]: Take this changes to the [xmippViz repo](https://github.com/I2PC/xmippViz).

File: software/em/xmipp/libraries/reconstruction/movie_alignment_correlation.cpp [hunk 4]
```diff
@@ -262,7 +262,7 @@ void ProgMovieAlignmentCorrelation::run()
 			gain.read(fnGain);
 			if (yDRcorner!=-1)
 				gain().selfWindow(yLTcorner, xLTcorner, yDRcorner, xDRcorner);
-			gain()=1.0/gain();
+			//Do not divide by the gain: gain()=1.0/gain();
```
Comment (dmaluenda) [inline software/em/xmipp/libraries/reconstruction/movie_alignment_correlation.cpp]: Take this change to the [Xmipp repo](https://github.com/I2PC/xmipp).

File: pyworkflow/em/packages/motioncorr/tests/test_protocols_motioncor2.py [hunk 5]
```diff
@@ -26,22 +26,24 @@
 # *
 # **************************************************************************
 
-from pyworkflow.em import ProtImportMovies
+from pyworkflow.em import ProtImportMovies, ImageHandler
```
Comment (dmaluenda) [inline pyworkflow/em/packages/motioncorr/tests/test_protocols_motioncor2.py]: Take this changes to the [Motioncorr plugin](https://github.com/scipion-em/scipion-em-motioncorr).

File: pyworkflow/em/packages/relion/dataimport.py [hunk 6]
```diff
@@ -202,10 +202,19 @@ def _findImagesPath(self, label, warnings=True):
...
+           
+            # Check if we have rot angle -> ALIGN_PROJ,
+            # if only psi angle -> ALIGN_2D
+            if row.containsLabel('rlnAngleRot') and row.getValue("rlnAngleRot") != 0.0:
+                self.alignType = ALIGN_PROJ
+            elif row.containsLabel('rlnAnglePsi') and row.getValue("rlnAnglePsi") != 0.0:
+                self.alignType = ALIGN_2D
+            else:
+                self.alignType = ALIGN_NONE
+
```
Comment (dmaluenda) [inline pyworkflow/em/packages/relion/dataimport.py]: Take this changes to the [Relion plugin](Take this changes to the [Motioncorr plugin](https://github.com/scipion-em/scipion-em-relion).

File: scipion [hunk 7]
```diff
@@ -41,9 +41,9 @@ except ImportError:
     from configparser import ConfigParser, ParsingError  # Python 3
 
 
-__version__ = ''
-__nickname__ = ''
-__releasedate__ = ''
+__version__ = 'v1.2.1'
+__nickname__ = 'Claudio'
+__releasedate__ = '2018-10-01'
```
Comment (dmaluenda) [inline scipion]: This change shouldn't be Pull Requested, isn't it??

File: pyworkflow/em/packages/xmipp3/protocol_particle_pick_pairs.py (hunk 1 above)
Comment (pconesa) [reply pyworkflow/em/packages/xmipp3/protocol_particle_pick_pairs.py]: Thanks @dmaluenda , I need to accelerate this to fix the java bug in devel-pluginization.

File: scipion (hunk 7 above)
Comment (pconesa) [reply scipion]: We usually, make a new commit correcting this once merged...is there a better way?

File: pyworkflow/em/packages/xmipp3/protocol_particle_pick_pairs.py (hunk 1 above)
Comment (dmaluenda) [reply pyworkflow/em/packages/xmipp3/protocol_particle_pick_pairs.py]: Done!

File: software/em/xmipp/applications/programs/movie_optical_alignment_cpu/movie_optical_alignment_cpu.cpp (hunk 2 above)
Comment (dmaluenda) [reply software/em/xmipp/applications/programs/movie_optical_alignment_cpu/movie_optical_alignment_cpu.cpp]: Done!

File: software/em/xmipp/java/src/xmipp/viewer/particlepicker/training/AutopickRunnable.java (hunk 3 above)
Comment (dmaluenda) [reply software/em/xmipp/java/src/xmipp/viewer/particlepicker/training/AutopickRunnable.java]: Done!

File: software/em/xmipp/libraries/reconstruction/movie_alignment_correlation.cpp (hunk 4 above)
Comment (dmaluenda) [reply software/em/xmipp/libraries/reconstruction/movie_alignment_correlation.cpp]: Done!

**Pull request merged**